### ISSUE TITLE
Skip state independent checks in prepare proposal

### ISSUE THREAD

**Issue by rootulp** [celestiaorg/celestia-app#3300]
## Context

https://github.com/celestiaorg/celestia-app/blob/1d73e334a48732ba2ef07caf064a7060175381f0/app/prepare_proposal.go#L36-L38

**Comment by rootulp** [comment]
It's not clear to me what the benefits are from doing this. It seems like a small performance improvement that we should only pursue if we have data indicating that:
- (a) prepare proposal takes too long and 
- (b) the ante handler step in prepare proposal is taking too much time relative to the rest of the operations

**Comment by evan-forbes** [comment]
it seems a bit risky to me as well. It's true that we've ran check tx successfully on all of the txs, however it seems like it would be easy to let txs slip through the cracks. For example, aren't we skipping signature verification when we call ReCheckTx? if that's the case, is it possible that a transaction gets included that invaldates the nonce of a tx that makes it to Prepare? What happens if a validator isn't running ReCheckTx?

we can quantify how long this is taking with further tracing and profiling, then re-evaluate.

**Comment by cmwaters** [comment]
> It's not clear to me what the benefits are from doing this. It seems like a small performance improvement that we should only pursue if we have data indicating that:

Yeah I agree that we should benchmark how long these operations take to get a better idea of the relative times. Maybe this is a good first step. We are benchmarking a lot of the network latencies as we're assuming that's were the biggest gains can be made, but it may also be a good idea to benchmark more of the computational work i.e. state execution, square construction, checkTx and the other abci calls as these do end up having significant impact on the total system (imagine waiting 3 seconds to process a block before you can continue to gossip it to peers that haven't seen the block yet).

**Comment by evan-forbes** [comment]
adding a plot here from a previous analysis, when the square size is increased, there are dramatic increases in the amount time required for nodes to process and prepare a proposal block. Worse yet, prepare and process proposal cannot overalp and are blocking, meaning that any increase in time verifying the data root is doubled.

This plot shows square sizes that are 512x512 ODS (anything greater than 32MB)

![method_duration_1-val](https://github.com/celestiaorg/celestia-app/assets/42654277/cda3735b-425a-4670-a958-45387f431d2f)

**Comment by rootulp** [comment]
Cool plot! IMO we need to drill into Prepare/ProcessProposal to see if the ante handler is a bottleneck before we do this issue.

**Comment by cmwaters** [comment]
Would be interested to see how much time the application is also handling `CheckTx` requests in a height
